libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -623,12 +631,13 @@ - (void)userChanged:(SFUserAccount *)user change:(SFUserAccountDataChange)change
 }
 
 - (void)notifyUserDataChange:(NSString *)notificationName withUser:(SFUserAccount *)user andChange:(SFUserAccountDataChange)change {
-
-    [[NSNotificationCenter defaultCenter] postNotificationName:notificationName
-                                                        object:user
-                                                      userInfo:@{
-                                                            SFUserAccountManagerUserChangeKey: @(change)
-                                                      }];
+    if (user) {
+        [[NSNotificationCenter defaultCenter] postNotificationName:notificationName
+                                                            object:user
+                                                          userInfo:@{
+                                                                SFUserAccountManagerUserChangeKey: @(change)
+                                                          }];
+    }
 
 }
```

Comment:
@khawkins  A Crash was reported by @sgoldberg-sfdc , he has added guards in S1 code to prevent these methods being called for nil user. These are additional checks on our end.